digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -181,10 +181,37 @@ public TaskResult runTask()
             AWSCredentials credentials = credentials(tag);
 
             SecretProvider awsSecrets = context.getSecrets().getSecrets("aws");
+            SecretProvider s3Secrets = awsSecrets.getSecrets("s3");
+            SecretProvider emrSecrets = awsSecrets.getSecrets("emr");
+            SecretProvider kmsSecrets = awsSecrets.getSecrets("kms");
+
+            Optional<String> emrEndpoint = Aws.first(
+                    () -> emrSecrets.getSecretOptional("endpoint"),
+                    () -> params.getOptional("endpoint", String.class));
```

Comment:
@komamitsu these endpoints are per-service as described [here](http://docs.aws.amazon.com/general/latest/gr/rande.html). Do you mean that we could use global AWS endpoint with per-service prefix like:

```java
awsSecrets.getSecretOptional("endpoint").transform(awsEndpoint -> "s3." + awsEndpoint)
...
awsSecrets.getSecretOptional("endpoint").transform(awsEndpoint -> "kms." + awsEndpoint)
```

? Glad to add this case if that is what you meant.